back/graphql_app/types/errors.py

```diff
@@ -0,0 +1,26 @@
+import strawberry
+
+
+@strawberry.interface
+class GraphQLError:
```

Comment:
```suggestion
class Error:
```

GraphQL 이라는 문맥은 충분히 공유가 되고 있는 것 같아서, 그냥 `Error` 로도 충분해 보입니다

Reply:
다른 앱이 생길 수도 있지 않을까 싶어서 Error로 해놓은 것도 있고, 클라이언트에 응답하기 이전에 발생할 수 있는 각 layer별 에러를 계층화해서 표현하면 서버 쪽의 오류인지 클라이언트 쪽의 오류인지 더 쉽게 파악할 수 있을 것 같아서 그렇게 구현했습니당!

Reply:
네임스페이스가 있기 때문에 다른 앱과 혼동될 일은 없을 것 같습니다..!

Reply:
앱을 나누냐 마냐 했던 저번 논의랑 비슷한 맥락인 것 같네요. 수정해주시거나 요청 주시면 제가 수정해서 merge 해놓겠읍니당 ㅎㅎㅎ

Reply:
네 수정은 제가 해도 될 것 같은데요. 앱을 나눈다고 해도 네임스페이스가 각자 있기 때문에 `A.Errror`, `B.Error` 이런 식으로 구분이 되기 때문에 안 겹쳐질 것 같다는 그런 생각이었습니다. 이게 만약에 없다면... 사실 이번 `Error` 뿐만 아니라 헷갈릴 수 있는 모든 클래스명과 모든 변수명에 앱 이름을 prefix 로 적던가 해야할텐데 그거는 좀 이상한 것 같아서요..!

Reply:
앱을 하나로 통일해서 작업해본 적이 없기도 없어서,, 괜찮으시다면 대략적인 호성님께서 가이드 라인을 따라서 다시 작업해보고 싶어요. 혹시 오늘 코멘트 말고 컨벤션 관련해서 더 도와주실 부분이 있으실까요? 항상 고맙고 죄송합니다 😢

Reply:
넵 나머지는 괜찮은 것 같습니다..! 에러 이름하고 QueryError, MutationError 를 일단 지우는게 어떨까 하는게 제 제안입니다

Reply:
그 아래의, 회원가입/로그인 mutation에서 발생할 수 있는 에러들을 계층화하고 상세 코드로 구분하는건 어떻게 생각하시나요?

Reply:
넵 필요하다면 만들어도 될 것 같습니다~ 상세 코드 같은 경우 같은 에러 타입 내에서 분기가 필요한 경우, 에러 계층화는 특정 에러을 확장하는 경우가 되겠네요. 상세 코드는 지금 도입해도 괜찮을 것 같고, 계층화는 아직은 좀 시기상조가 아닌가 생각합니다.

Reply:
프로젝트 기간이 좀 짧기도 하고, 호성님께서 알려주신 웹 graphql 클라이언트를 보다 보니까 어떻게든 추가 문서 없이 docstring만으로 뜻을 전달할 수 있으면 좋을 것 같다는 생각이 들었었읍니당 ㅎ 어렵네요,,,,,,,

Reply:
넵넵 맞아요. 코드는 확실히 도움이 될 것 같습니다. 계층화는 필요한 케이스가 나오면 그 때 확장하는게 코드 복잡도가 낮아진다고 생각합니다..!

Reply:
이때까지 그래왔듯 제가 그 정도를 잘 몰라서,,,, 지금까지는 항상 투머치였던 것 같네요 이번주 많이 바쁘신가요?

Reply:
프로젝트 기간이 짧지만 이런 논의를 통해서 여러가지 배울 수 있어서 좋은 것 같습니다